mono/metadata/icall.c

```diff
@@ -5740,26 +5740,30 @@ ves_icall_Mono_Runtime_ExceptionToState (MonoExceptionHandle exc_handle, guint64
 	MonoStringHandle result;
 
 #ifndef DISABLE_CRASH_REPORTING
-	// FIXME: Push handles down into mini/mini-exceptions.c
-	MonoException *exc = MONO_HANDLE_RAW (exc_handle);
-	MonoThreadSummary out;
-	mono_get_eh_callbacks ()->mono_summarize_exception (exc, &out);
-
-	*portable_hash_out = (guint64) out.hashes.offset_free_hash;
-	*unportable_hash_out = (guint64) out.hashes.offset_rich_hash;
-
-	JsonWriter writer;
-	mono_json_writer_init (&writer);
-	mono_native_state_init (&writer);
-	gboolean first_thread_added = TRUE;
-	mono_native_state_add_thread (&writer, &out, NULL, first_thread_added, TRUE);
-	char *output = mono_native_state_free (&writer, FALSE);
-	result = mono_string_new_handle (mono_domain_get (), output, error);
-	g_free (output);
+	if (mono_get_eh_callbacks ()->mono_summarize_exception) {
+		// FIXME: Push handles down into mini/mini-exceptions.c
+		MonoException *exc = MONO_HANDLE_RAW (exc_handle);
+		MonoThreadSummary out;
+		mono_get_eh_callbacks ()->mono_summarize_exception (exc, &out);
+
+		*portable_hash_out = (guint64) out.hashes.offset_free_hash;
+		*unportable_hash_out = (guint64) out.hashes.offset_rich_hash;
+
+		JsonWriter writer;
+		mono_json_writer_init (&writer);
+		mono_native_state_init (&writer);
+		gboolean first_thread_added = TRUE;
+		mono_native_state_add_thread (&writer, &out, NULL, first_thread_added, TRUE);
+		char *output = mono_native_state_free (&writer, FALSE);
+		result = mono_string_new_handle (mono_domain_get (), output, error);
+		g_free (output);
+	} else 
 #else
-	*portable_hash_out = 0;
-	*unportable_hash_out = 0;
-	result = mono_string_new_handle (mono_domain_get (), "", error);
+	{
+		*portable_hash_out = 0;
+		*unportable_hash_out = 0;
+		result = mono_string_new_handle (mono_domain_get (), "", error);
+	}
 #endif
```

Comment:
shouldn't the `#endif` move up and replace `#else`?

Reply:
There, I moved the return up into the if block, and moved it all into a single if, with a fallthrough for the else.